lib/src/feeds.dart

```diff
@@ -32,41 +31,48 @@ class FeedsPage extends StatefulWidget {
 class _FeedsPageState extends State<FeedsPage> {
   bool _disposed = false;
 
+  List<Feed> feeds;
+  StreamSubscription sub;
+
   @override
   void initState() {
     super.initState();
 
-    if (feedManager.feeds == null) {
-      feedManager.load(context).then((_) {
-        if (!_disposed) {
-          setState(() {
-            //
-          });
-        }
-      });
-    }
+    this.feeds = feedManager.feeds;
+
+    feedManager.onFeedsChanged.listen((feeds) {
+      if (!_disposed) {
+        setState(() {
+          this.feeds = feeds;
+        });
+      }
+    });
   }
 
   @override
   void dispose() {
     super.dispose();
 
+    sub?.cancel();
+
     _disposed = true;
   }
 
   @override
   Widget build(BuildContext context) {
     Widget body;
 
-    if (feedManager.error != null) {
-      return new Scaffold(
-        appBar: new AppBar(
-          title: new Text(FeedsPage.title),
-          automaticallyImplyLeading: false,
-        ),
-        body: new Center(child: new Text(feedManager.error)),
-      );
-    } else if (feedManager.feeds == null) {
+    if (feedManager.feeds == null) {
+      feedManager.load().catchError((e) {
```

Comment:
The function passed to catchError is repeated started on line 96.

Reply:
gotcha. it's small enough that I'm ok with the code dup